include/boost/url/experimental/detail/router.hpp

```diff
@@ -0,0 +1,205 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_EXPERIMENTAL_DETAIL_ROUTER_HPP
+#define BOOST_URL_EXPERIMENTAL_DETAIL_ROUTER_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/decode_view.hpp>
+#include <boost/url/variant.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/grammar/all_chars.hpp>
+#include <boost/url/grammar/alnum_chars.hpp>
+#include <boost/url/grammar/alpha_chars.hpp>
+#include <boost/url/grammar/lut_chars.hpp>
+#include <boost/url/grammar/optional_rule.hpp>
+#include <boost/url/grammar/token_rule.hpp>
+#include <boost/url/grammar/variant_rule.hpp>
+#include <boost/url/grammar/not_empty_rule.hpp>
+#include <boost/url/grammar/detail/charset.hpp>
+#include <boost/url/rfc/pchars.hpp>
+#include <boost/url/rfc/detail/path_rules.hpp>
+#include <boost/core/ignore_unused.hpp>
+#include <vector>
+#include <locale>
+
+namespace boost {
+namespace urls {
+namespace experimental {
+namespace detail {
+
+struct match_any_t {};
+
+// A dynamic path segment
+class segment_template
+{
+    std::string str_;
+    bool is_literal_ = true;
+
+    friend struct segment_template_rule_t;
+public:
+    segment_template() = default;
+
+    bool
+    match(pct_string_view seg) const
+    {
+        if (is_literal_)
+            return *seg == str_;
+
+        return true;
+    }
+
+    string_view
+    string() const
+    {
+        return str_;
+    }
+
+
+    string_view
+    id() const
```

Comment:
The id is just a comment, what is this function for?

Reply:
This will be used in the match results very soon.

I talked to Klemens about his use cases. He also told me to read the openAPI spec.

I'm convinced the ability to get the value of a named parameters is very important.

Reply:
Klemens' use-case is in formatting not routing

Reply:
Yes. That's been considered.